src/tasks/MonoTargetsTasks/ILStrip/ILStrip.cs

```diff
@@ -12,45 +12,95 @@
 using CilStrip.Mono.Cecil.Binary;
 using CilStrip.Mono.Cecil.Cil;
 using CilStrip.Mono.Cecil.Metadata;
+using System.Reflection.Metadata;
+using System.Reflection.Metadata.Ecma335;
+using System.Reflection.PortableExecutable;
 
 public class ILStrip : Microsoft.Build.Utilities.Task
 {
     /// <summary>
     /// Assemblies to be stripped.
     /// The assemblies will be modified in place if OutputPath metadata is not set.
     /// </summary>
-    [Required]
     public ITaskItem[] Assemblies { get; set; } = Array.Empty<ITaskItem>();
 
     /// <summary>
     /// Disable parallel stripping
     /// </summary>
     public bool DisableParallelStripping { get; set; }
 
+    /// <summary>
+    /// Enable the feature of trimming indiviual methods
+    /// </summary>
+    public bool TrimIndividualMethods { get; set; }
+
+    /// <summary>
+    /// Methods to be trimmed, identified by method token
+    /// </summary>
+    public ITaskItem[] MethodTokenFiles { get; set; } = Array.Empty<ITaskItem>();
+
+    /// <summary>
+    /// Directory where all the assemblies are stored
+    /// </summary>
+    public string AssemblyPath { get; set; } = "";
```

Comment:
Why is it needed to specify path-to-assemblies explicitly? 
`Assemblies` should already include the list of all assemblies (and their absolute paths).

Reply:
Additionally, items in `Assemblies` list, could include metadata specifying the location of the MethodTokenFile, instead of including all `*.txt` from a specific folder. The metadata could be populated in the MonoAOTCompiler task.

Reply:
Sorry, I've might not been clear enough and might be wrong, but my understanding is that ILStrip would not be used as a standalone build task, but rather integrated with MonoAOTCompiler.

Reply:
Actually, I removed `AssemblyPath` after I used `ass->image->filename` instead.

Reply:
I don't think that it is necessary to tag ILStrip task along with MonoAOTCompiler task. One could also trigger aot compile of an assembly by manually run the mono aot compile command.

Reply:
That is true, however, how would the stripping be initiated in that manual setup?

Reply:
xamarin-macios doesn't use the MonoAOTCompiler task yet but it does use ILStrip so it might be useful to have the two uncoupled.

Reply:
I just wanted to point out that some bundler task (Xamarin or not) would/could set up something like:
```
obj/Release/net8.0-ios/ios-arm64/linked/MonoMauiiOS.dll
    AOTAssembly = obj/Release/net8.0-ios/ios-arm64/nativelibraries/aot-output/arm64/MonoMauiiOS.dll.s
    Arch = arm64
    ...
    ProcessArguments = --debug --llvm -O=gsharedvt
    ObjectFile = obj/Release/net8.0-ios/ios-arm64/nativelibraries/aot-output/arm64/MonoMauiiOS.dll.o
    Abi = ARM64+LLVM
    LLVMFile = obj/Release/net8.0-ios/ios-arm64/nativelibraries/aot-output/arm64/MonoMauiiOS.dll.llvm.o
    AOTData = obj/Release/net8.0-ios/ios-arm64/nativelibraries/aot-output/arm64/MonoMauiiOS.aotdata
```
This is one of the `@(_AssembliesToAOT)` items with all its metadata from Xamarin `AOTCompile` build task. 

If we impose a requirement that assemblies that get passed as inputs to ILStrip task need to have a required metadata set, eg: `MethodTokenFilePathForAssembly` I believe it would simplify the code and integration process. 

The example above would become:
```
obj/Release/net8.0-ios/ios-arm64/linked/MonoMauiiOS.dll
    AOTAssembly = obj/Release/net8.0-ios/ios-arm64/nativelibraries/aot-output/arm64/MonoMauiiOS.dll.s
    Arch = arm64
    ...
    ProcessArguments = --debug --llvm -O=gsharedvt
    ObjectFile = obj/Release/net8.0-ios/ios-arm64/nativelibraries/aot-output/arm64/MonoMauiiOS.dll.o
    Abi = ARM64+LLVM
    LLVMFile = obj/Release/net8.0-ios/ios-arm64/nativelibraries/aot-output/arm64/MonoMauiiOS.dll.llvm.o
    AOTData = obj/Release/net8.0-ios/ios-arm64/nativelibraries/aot-output/arm64/MonoMauiiOS.aotdata
    MethodTokenFilePathForAssembly=obj/Release/net8.0-ios/ios-arm64/nativelibraries/aot-output/arm64/MonoMauiiOS.aoted_methods.txt
```

Reply:
Added metadata MethodTokenFile to CompiledAssemblies

Reply:
@akoeplinger This new feature shouldn't affect how xamarin-macios uses ILStrip.